master/pkg/tasks/task_gc.go

```diff
@@ -46,6 +46,10 @@ func (g GCCkptSpec) ToTaskSpec(allocationToken string) TaskSpec {
 	}
 	res.Environment = schemas.WithDefaults(env).(expconf.EnvironmentConfig)
 
+	resSpec := res.Environment.PodSpec()
+	newSpec := resSpec.Merge(g.LegacyConfig.PodSpec()).(*expconf.PodSpec)
+	res.Environment.RawPodSpec = newSpec
```

Comment:
Don't use the `Raw` API here, use the `SetPodSpec()`.  The Raw API is only for when you have not finished parsing something.